default-templates/primary-ignition.tmpl

```diff
@@ -10,22 +10,29 @@ systemd:
     - name: etcd2.service
       enable: true
       dropins:
-       - name: 10-etcd2-kocho.conf
+       - name: 30-etcd2-kocho.conf
          contents: |
-            [Service]
-            Environment="ETCD_DISCOVERY={{.DiscoveryUrl}}"
-            Environment="ETCD_ADVERTISE_CLIENT_URLS=http://$private_ipv4:2379"
-            Environment="ETCD_INITIAL_ADVERTISE_PEER_URLS=http://$private_ipv4:2380"
-            Environment="ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379"
-            Environment="ETCD_LISTEN_PEER_URLS=http://$private_ipv4:2380,http://127.0.0.1:2380"
-            Environment="ETCD_ELECTION_TIMEOUT=5000"
+           [Unit]
+           Requires=coreos-metadata.service
+           After=coreos-metadata.service
+
+           [Service]
+           EnvironmentFile=/run/metadata/coreos
+           ExecStart=
+           ExecStart=/usr/bin/etcd2 \
+               --discovery={{.DiscoveryUrl}} \
+               --advertise-client-urls=http://${COREOS_EC2_IPV4_LOCAL}:2379 \
```

Comment:
It comes with the latest versions and it is normally used by ignition in such a cases